src/interfaces/web3-connection-options.ts

```diff
@@ -16,6 +17,11 @@ export interface Web3ConnectionOptions {
    * @note you can provide a node server if you're using IPC
    */
   web3ProviderOptions?: HttpProviderOptions | WebsocketProviderOptions;
+  
+  /**
+   * Pass a custom provider instead
+   */
+  web3CustomProvider?: Provider;
```

Comment:
When assigning the WalletConnectProvider` to `web3CustomProvider typescript throws: Type 'WalletConnectProvider' is not assignable to type 'provider | undefined'.